### ISSUE TITLE
Esp32 How to use mqtt_ As to connect the brokers of two servers？

### ISSUE THREAD

**Issue by oaux** [peterhinch/micropython-mqtt#113]
I tried to create two MQTTclients, but many wifi error prompts

**Comment by peterhinch** [comment]
I don't think this can work. You would have two sockets (one on each client) connected to the same port. There is no mechanism for determining the correct destination socket for an incoming packet. You might be able to do something if you could ensure that the two brokers used different ports.

**Comment by oaux** [comment]
If two brokers use different ports, how to handle it? I tried to hide the WiFi reconnection code, but it would cause the two links to switch repeatedly

**Comment by peterhinch** [comment]
Good point. I think it would involve a lot of work. The module was designed on the basis of a one-to-one correspondence between client and broker.

**Comment by gunbyl** [comment]
I have tested to have two client in a ESP32-S3 and the test application works for days of testing without any errors! A good thing about ESP32-S3 is that it has about 8 Mb of memory.  One client is connected to Bevywise MQTT and the other client is connected to ThingSpeak MQTT.  I have two configs, two event driven messages routines, two up routines etc. Only the client connected to Bevywise has a config with wifi credentials included as there is only one common wifi for both clients of cource. I also start the two clients as two separate tasks. I have used Peters 2.3 Example Usage from the README doc as a starting point.

**Comment by peterhinch** [comment]
Interesting. Thanks for the report.
